gobblin-service/src/main/java/org/apache/gobblin/service/modules/flow/BaseFlowToJobSpecCompiler.java

```diff
@@ -181,6 +190,19 @@ private  AddSpecResponse onAddFlowSpec(FlowSpec flowSpec) {
 
     // always try to compile the flow to verify if it is compilable
     Dag<JobExecutionPlan> dag = this.compileFlow(flowSpec);
+
+    if (this.warmStandbyEnabled &&
+        (!flowSpec.getConfigAsProperties().containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY) || PropertiesUtils.getPropAsBoolean(flowSpec.getConfigAsProperties(), ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false"))) {
+      try {
+        for (Dag.DagNode<JobExecutionPlan> dagNode : dag.getStartNodes()) {
+          userQuotaManager.checkQuota(dagNode);
```

Comment:
I hope we can make the check for all starting nodes atomic as well. So any node fail the check, we need to rollback the previous one